### PR TITLE
simpler reference test for 11p

### PR DESCRIPTION
**Pull request by vene**
in the simple, pseudocode like reference implementation we **do not** have to skip the (p=1, r=0) point, but we have to skip it in the clever numpy implementation in the main code.

> Why?

I have no idea.

> Should it actually be skipped in the simple reference implementation?

I'm not sure. If it's not skipped, the first of the 11 points is always 1, because max. precision at R=0 is always 1. So I'm inclined to think that it should actually be skipped, and that the "fast" implementation is incorrect.

Look at the example plot fig. 8.3 in [the ir book](https://nlp.stanford.edu/IR-book/html/htmledition/evaluation-of-ranked-retrieval-results-1.html) (the rotated one).

It is a "micro-averaged" interpolated average precision *curve* over multiple queries, i.e., the point on the curve at recall=r is mean over queries of max(precision[recall >= r])

The relationship between it and interpolated average precision is: interpolated average precision takes this curve over a single query, and averages its y values.

The curve in fig. 8.3 is not exactly 1 at recall=0, suggesting that the interpolated precision at recall=0 is not always 1. With our implementation, it seems like it always is.

### CODE REVIEW

File: sklearn/metrics/tests/test_ranking.py [hunk 1]
```diff
@@ -13,13 +13,11 @@
...
-from sklearn.utils.testing import assert_equal
-from sklearn.utils.testing import assert_almost_equal
-from sklearn.utils.testing import assert_array_equal
-from sklearn.utils.testing import assert_array_almost_equal
-from sklearn.utils.testing import assert_warns
+from sklearn.utils.testing import (assert_raises, clean_warning_registry,
+                                   assert_raise_message, assert_equal,
+                                   assert_almost_equal, assert_array_equal,
+                                   assert_array_almost_equal, assert_warns,
+                                   assert_allclose)
```
Comment (GaelVaroquaux) [inline sklearn/metrics/tests/test_ranking.py:20]: Why this change? It touches a lot of lines for little add value

Comment (vene) [reply sklearn/metrics/tests/test_ranking.py:20]: the code was using assert_array_almost_equal to compare scalars... I can revert it, but let me know what you think about the pr description (just edited)

File: sklearn/metrics/tests/test_ranking.py [hunk 2]
```diff
@@ -512,14 +521,15 @@ def _test_precision_recall_curve(y_true, probas_pred):
...
         y_true, probas_pred)
-    assert_array_almost_equal(precision_recall_auc, 0.859, 3)
-    assert_array_almost_equal(precision_recall_auc,
-                              average_precision_score(y_true, probas_pred))
-    assert_equal(interpolated_average_precision,
-                 average_precision_score(y_true, probas_pred,
-                                         interpolation='eleven_point'))
-    assert_almost_equal(_average_precision(y_true, probas_pred),
-                        precision_recall_auc, decimal=3)
+    print(precision_recall_auc)
```
Comment (GaelVaroquaux) [inline sklearn/metrics/tests/test_ranking.py:524]: Trailing print that you should remove

**Comment by vene** [comment]
closing this PR, I will just leave a note on your PR explaining what I think the issue is
